src/Compilers/CSharp/Portable/FlowAnalysis/FlowAnalysisPass.cs

```diff
@@ -41,9 +47,27 @@ internal class FlowAnalysisPass
                 // will be reported by the lambda binder.
                 Debug.Assert(method.MethodKind != MethodKind.AnonymousFunction);
 
+                // Add implicit "return default(T)" if this is a submission that does not have a trailing expression.
+                var submissionResultType = (method as SynthesizedInteractiveInitializerMethod)?.ResultType;
+                if (!hasTrailingExpression && ((object)submissionResultType != null))
+                {
+                    Debug.Assert(submissionResultType.SpecialType != SpecialType.System_Void);
+
+                    var trailingExpression = new BoundDefaultOperator(method.GetNonNullSyntaxNode(), submissionResultType);
+                    var newStatements = block.Statements.Add(new BoundReturnStatement(trailingExpression.Syntax, trailingExpression));
+                    block = new BoundBlock(block.Syntax, ImmutableArray<LocalSymbol>.Empty, newStatements) { WasCompilerGenerated = true };
+#if DEBUG
+                    // It should not be necessary to repeat analysis after adding this node, because adding a trailing
+                    // return in cases where one was missing should never produce different Diagnostics.
+                    var flowAnalysisDiagnostics = DiagnosticBag.GetInstance();
+                    Debug.Assert(!Analyze(compilation, method, block, flowAnalysisDiagnostics));
+                    Debug.Assert(flowAnalysisDiagnostics.ToReadOnly().SequenceEqual(diagnostics.ToReadOnly()));
```

Comment:
No, good point...  Let's see how I can do this without creating another ```DiagnosticBag``` in release...